File: swift/LongestCommonPrefix.swift
// https://leetcode.com/problems/longest-common-prefix/

class Solution {
    func longestCommonPrefix(_ strs: [String]) -> String {
        if strs.count == 0 { return "" }
       
        var minStr = (strs.min { $0.count < $1.count })!
        
        for s in strs {
            while !s.hasPrefix(minStr) {
                 minStr.removeLast()
            }
        }
        
        return minStr 
    }
}

File: python/LongestCommonPrefix.py
# https://leetcode.com/problems/longest-common-prefix/

class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        prefix = strs[0]
        for s in strs:
            count = 0
            for i in range(min(len(prefix), len(s))):
                if s[i] != prefix[i]:
                    break
                else:
                    count += 1
            prefix = prefix[0:count]
        return prefix